changelog/fragments/k8s-1.24-bump.yaml

```diff
@@ -0,0 +1,144 @@
+
+        1. In the `go.mod` file:
+
+        - Replace `go 1.17` with `go 1.18`   
+
+        2. Run `go mod tidy` to ensure all dependencies are upgraded
+
+  - description: >
+      For `operator-sdk bundle validate` command and GoodPractices validator (`--select-optional name=good-practices`): fix channel naming validation [More info](https://github.com/operator-framework/api/pull/231)
+      
+    # kind is one of:
+    # - addition
+    # - change
+    # - deprecation
+    # - removal
+    # - bugfix
+    kind: "bugfix"
+
+    # Is this a breaking change?
+    breaking: false
+
+  - description: >
+      For `operator-sdk bundle validate` command and GoodPractices validator (`--select-optional name=good-practices`): Add CRD description check validation [More info](https://github.com/operator-framework/api/pull/234)
+      
+    # kind is one of:
+    # - addition
+    # - change
+    # - deprecation
+    # - removal
+    # - bugfix
+    kind: "addition"
+
+    # Is this a breaking change?
+    breaking: false
+
+  - description: >
+      For `operator-sdk bundle validate` command and GoodPractices validator (`--select-optional name=good-practices`): Add a new check to warn authors when permissions to create CRDs are found [More info](https://github.com/operator-framework/api/pull/241)
+      
+    # kind is one of:
+    # - addition
+    # - change
+    # - deprecation
+    # - removal
+    # - bugfix
+    kind: "addition"
+
+    # Is this a breaking change?
+    breaking: false
+
+  - description: >
+      For Helm (helm/v1) and Hybrid Helm (hybrid.helm/v1-alpha) language based operators: Upgrade helm from `v3.6.2` to `v3.9.0`
+    breaking: false
+
+  - description: >
+      For Ansible (ansible/v1) and Helm (helm/v1) language based operators: change the default value for the manager.option LeaderElectionResourceLock from `comfigmaps` to `configmapsleases` to allow a safe transition to use the leases option on the future. [More info](https://github.com/kubernetes-sigs/controller-runtime/pull/1903)
+    breaking: false
+  
+  - description: >
+      For Quarkus (quarkus/v1-alpha) language based operator: add bundle generation [More info](https://github.com/operator-framework/java-operator-plugins/pull/84)
+    breaking: false
+  
+  - description: >
+      For Quarkus (quarkus/v1-alpha) language based operator: upgrade dependency from `v0.0.3` to `v0.5.1` [More info](https://github.com/operator-framework/java-operator-plugins/releases)
+    breaking: false
+  
+  - description: >
+      For Helm Hybrid (hybrid.helm/v1-alpha) language based operator: Upgrade dependency from `v0.0.10` to `v0.0.11`. [More info](https://github.com/operator-framework/helm-operator-plugins/releases)
+    breaking: false
+
+  - description: >
+      For Go (go/v3, go/v2), Ansible (ansible/v1), Helm (helm/v1), and Helm Hybrid (hybrid.helm/v1-alpha) language based operators: Bump OPM version from `v1.19.1` to `v1.23.0`
```

Comment:
optional nit: replace OPM with operator-registry/opm (I'm not confident this is necessary or correct though :)

Reply:
operator-registry/opm -> dep is used internally only for the run bundle and bundle-upgrade
These docs are for end-users they do not care about what deps SDK CLI uses. 
They care about what changes in the release affected them
So, if we look at the release log of operator-registry/opm and check changes/fixes that affect these command then, ihmo we need call out here something like fix issue x when run command y [More info](PR)

Reply:
In this case the only changes relevants for the commands and for SDK users are : 

* ✨ improve message from common scenario faced to add a new bundle by [@camilamacedo86](https://github.com/camilamacedo86) in [#954](https://github.com/operator-framework/operator-registry/pull/954)
* ✨ adds some extra text context to sql statement errors by [@camilamacedo86](https://github.com/camilamacedo86) in [#953](https://github.com/operator-framework/operator-registry/pull/953)

@everettraven could we create a new entry here with something like:

```
  - description: >
      For operator-sdk run bundle and bundle-upgrade subcommands: improve message for failures scenarios and add extra context for errors faced when them are using an SQLIndex (old format). More info:  [#953](https://github.com/operator-framework/operator-registry/pull/953) and [#954](https://github.com/operator-framework/operator-registry/pull/954)
    breaking: false
```

Reply:
@camilamacedo86 should be updated as of commit https://github.com/operator-framework/operator-sdk/pull/5843/commits/9a5fbdd6ca84c05b4513f8df77ee5b50b88169ac